plutus-contract/src/Language/Plutus/Contract/StateMachine.hs

```diff
@@ -161,7 +162,9 @@ getOnChainState ::
 getOnChainState StateMachineClient{scInstance, scChooser} = mapError (review _SMContractError) $ do
     utxo <- utxoAt (SM.machineAddress scInstance)
     let states = getStates scInstance utxo
-    either (throwing _SMContractError) (\s -> pure (s, utxo)) (scChooser states)
+    case states of
+        [] -> throwError NoOnChainState
```

Comment:
Maybe we should just return a maybe here? Arguably this is more of a "normal" return than an error? Not sure.

Reply:
Not sure either. It's sort of handling a special case, not all contracts would want to handle it.

Reply:
So I guess the other errors aren't really "recoverable" ones. I think in this case it would be sensible to return a maybe - the user can just throw immediately if they can't handle that.

Reply:
And arguably "one participant starts trying to do things before the machine has started" is actually a very common case that users *should* handle. Making it an error makes it easy to just "let it crash" in this case, which probably isn't right.

Reply:
Fair enough.